protocol/app/module/interface_registry.go

```diff
@@ -0,0 +1,100 @@
+package module
+
+import (
+	"cosmossdk.io/x/tx/signing"
+	"fmt"
+	addresscodec "github.com/cosmos/cosmos-sdk/codec/address"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	gogoproto "github.com/cosmos/gogoproto/proto"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+var InterfaceRegistry types.InterfaceRegistry
+
+func getLegacyMsgSignerFn(path []string) func(msg proto.Message) ([][]byte, error) {
+	if len(path) == 0 {
+		panic("path is expected to contain at least one value.")
+	}
+
+	return func(msg proto.Message) ([][]byte, error) {
+		m := msg.ProtoReflect()
+		for _, p := range path[:len(path)-1] {
+			fieldDesc := m.Descriptor().Fields().ByName(protoreflect.Name(p))
+			if fieldDesc.Kind() != protoreflect.MessageKind {
+				return nil, fmt.Errorf("Expected for field %s to be Message type in path %+v for msg %+v.", p, path, msg)
+			}
+			v := m.Get(fieldDesc)
+			if !v.IsValid() {
+				return nil, fmt.Errorf("Expected for field %s to be populated in path %+v for msg %+v.", p, path, msg)
+			}
+			m = v.Message()
+		}
+
+		fieldDesc := m.Descriptor().Fields().ByName(protoreflect.Name(path[len(path)-1]))
+		if fieldDesc.Kind() != protoreflect.StringKind {
+			return nil, fmt.Errorf(
+				"Expected for final field %s to be String type in path %+v for msg %+v.",
+				path[len(path)-1],
+				path,
+				msg,
+			)
+		}
+		signer, err := sdk.AccAddressFromBech32(m.Get(fieldDesc).String())
+		if err != nil {
+			return nil, err
+		}
+		return [][]byte{signer}, nil
+	}
+}
+
+// TODO(CORE-538): Consider having app injected messages return an error instead of empty signers list.
+func noSigners(_ proto.Message) ([][]byte, error) {
+	return [][]byte{}, nil
+}
+
+func init() {
+	var err error
+
+	if InterfaceRegistry, err = NewInterfaceRegistry(
+		sdk.GetConfig().GetBech32AccountAddrPrefix(),
+		sdk.GetConfig().GetBech32ValidatorAddrPrefix(),
+	); err != nil {
+		panic(err)
+	}
+}
+
+func NewInterfaceRegistry(addrPrefix string, valAddrPrefix string) (types.InterfaceRegistry, error) {
```

Comment:
They have coverage via e2e test framework since the framework signs all messages that go through the whole ante handler stack.

Also, this will go away with Cosmos ?0.50.3? since nesting has been [fixed](https://github.com/cosmos/cosmos-sdk/issues/18722) and we can migrate to just using the annotations in all situations.